src/order/succ_pred/basic.lean

```diff
@@ -220,10 +220,21 @@ by rw [←Ici_inter_Iic, Ici_succ_of_not_is_max ha, Ioi_inter_Iic]
 lemma Ico_succ_left_of_not_is_max (ha : ¬ is_max a) : Ico (succ a) b = Ioo a b :=
 by rw [←Ici_inter_Iio, Ici_succ_of_not_is_max ha, Ioi_inter_Iio]
 
+/-- A successor limit is a value that isn't a successor. -/
+def is_succ_limit (b : α) : Prop :=
+∀ a, succ a ≠ b
```

Comment:
Isn't `not_is_succ_limit_of_is_max` a weird theorem? Surely `⊤` is supposed to be a limit in `with_top ℕ`? Maybe you want to change the definition to
```suggestion
def is_succ_limit (b : α) : Prop := ∀ a, succ a = b → is_max a
```

Reply:
Yeah, that seems more sensible. On it.